src/main/java/org/craftercms/studio/controller/rest/v2/PublishController.java

```diff
@@ -215,6 +207,23 @@ public ResponseBody getAvailablePublishingTargets(@RequestParam(name = REQUEST_P
         return responseBody;
     }
 
+    @ValidateParams
+    @GetMapping(value = HAS_INITIAL_PUBLISH, produces = APPLICATION_JSON_VALUE)
+    public ResponseBody hasInitialPublish(@EsapiValidatedParam(type = SITE_ID) @RequestParam(name = REQUEST_PARAM_SITEID) String siteId)
+        throws SiteNotFoundException {
+        if (!siteService.exists(siteId)) {
```

Comment:
Let's throw these exceptions from the service layer
Let's use
```siteService.checkSiteExists(siteId);```

Reply:
Moved check site exists to services layer. Thanks.